lib/mssql.js

```diff
@@ -342,15 +342,39 @@ function buildLimit(limit, offset) {
   return sql;
 }
 
+MsSQL.prototype.buildPartitionBy = function(model, where) {
+  if (!where) {
+    return '';
+  }
+
+  if (typeof where !== 'object' || Array.isArray(where)) {
+    return '';
+  }
+
+  for (var key in where) {
+    if (key === 'and' || key === 'or') {
+      return '';
+    }
+    var columnName = this.columnEscaped(model, key);
+    break;
+  }
+
+  return 'PARTITION BY ' + columnName;
+};
+
 MsSQL.prototype.buildColumnNames = function(model, filter) {
   var columnNames = this.invokeSuper('buildColumnNames', model, filter);
   if (filter.limit || filter.offset || filter.skip) {
     var orderBy = this.buildOrderBy(model, filter.order);
     var orderClause = '';
+    var partitionByClause = '';
+    if(filter.where) {
+      partitionByClause = this.buildPartitionBy(model, filter.where);
+    }
     if (orderBy) {
-      orderClause = 'OVER (' + orderBy + ') ';
+      orderClause = 'OVER (' + partitionByClause + ' ' + orderBy + ') ';
```

Comment:
I'm not sure why you need to replace `ORDER BY` with `PARTITION BY`. The purpose was to support pagination for early version of SQL servers that don't support `OFFSET-FETCH` clause. See http://www.sqlshack.com/sql-server-2012-introduction-pagination/.